core/src/main/java/io/questdb/cairo/ColumnVersionWriter.java

```diff
@@ -102,6 +112,23 @@ public void truncate() {
         }
     }
 
+    public void upsert(long timestamp, ColumnVersionReader detachedColumnVersionReader) {
+        LongList detachedEntries = detachedColumnVersionReader.getCachedList();
+        LongList entries = getCachedList();
+        for (int i = 0, limit = detachedEntries.size(); i < limit; i += 4) {
```

Comment:
Can we insert all records on one go? Both lists are sorted by timestamp and then column index

Reply:
I have done my best. For new columns it is easier to upsert upfront, otherwise I have made the change.